torch/_dynamo/variables/builder.py

```diff
@@ -620,6 +627,16 @@ def build_key_value(i, k, v):
                 else GuardBuilder.TYPE_MATCH
             )
             return NumpyVariable(value, source=self.source)
+        elif is_numpy_dtype(value):
+            self.install_guards(GuardBuilder.ID_MATCH)
+            return NumpyDTypeVariable(value, source=self.source)
+        elif is_numpy_type_info(value):
+            if isinstance(value, np.iinfo):
+                dt_source = AttrSource(self.source, "dtype")
+                install_guard(dt_source.make_guard(GuardBuilder.ID_MATCH))
+            else:
+                self.install_guards(GuardBuilder.ID_MATCH)
```

Comment:
@lezcano Test added, this patten seems to resolve the issue. `np.iinfo` objects don't implement equality and are not cached on creation like `np.dtype` and `np.finfo`.

Reply:
This is almost correct. You also need to add a type guard onto self, otherwise you might error out on the `.dtype` access. But otherwise I like this pattern! Much cleaner than what I proposed.